src/libcollections/vec.rs

```diff
@@ -427,8 +428,12 @@ impl<T> Vec<T> {
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn as_mut_slice<'a>(&'a mut self) -> &'a mut [T] {
         unsafe {
+            let ptr = *self.ptr;
+            if cfg!(not(stage0)) { // NOTE remove cfg after next snapshot
+                assume(ptr != 0 as *mut T);
```

Comment:
Actually, using `is_null()` works now, that didn't work the first time I tried, either because of an error on my side or because of the `ptrtoint` instruction it was generated with the old `is_null()` implementation.
Thanks!